homeassistant/components/husqvarna_automower/calendar.py

```diff
@@ -59,56 +73,23 @@ async def async_get_events(
 
         This is only called when opening the calendar in the UI.
         """
-        self.calendar = await hass.async_add_executor_job(Calendar)
-        self._automower_to_ical_event(self.mower_attributes.calendar.events)
-        events = self.calendar.timeline_tz(start_date.tzinfo).overlapping(
+        schedule = self.mower_attributes.calendar
+        if schedule.timeline is None:
+            raise HomeAssistantError("Unable to get events: No schedule set")
+        cursor = schedule.timeline.overlapping(
             start_date,
             end_date,
         )
-        return [_ical_to_calendar_event(event) for event in events]
-
-    def _automower_to_ical_event(
-        self, event_list: list[AutomowerCalendarEvent]
-    ) -> None:
-        """Convert the automower event to an ical event an store it."""
-        schedule_no: dict = {}
-        for event in event_list:
-            if event.work_area_id is not None:
-                schedule_no[event.work_area_id] = 0
-            if event.work_area_id is None:
-                schedule_no["-1"] = 0
-        for event in self.mower_attributes.calendar.events:
-            wa_name = ""
-            if event.work_area_id is not None:
-                if self.mower_attributes.work_areas is not None:
-                    _work_areas = self.mower_attributes.work_areas
-                    wa_name = f"{_work_areas[event.work_area_id].name} "
-                    schedule_no[event.work_area_id] = (
-                        schedule_no[event.work_area_id] + 1
-                    )
-                    number = schedule_no[event.work_area_id]
-            if event.work_area_id is None:
-                schedule_no["-1"] = schedule_no["-1"] + 1
-                number = schedule_no["-1"]
-            self.calendar.events.append(
-                Event(
-                    dtstart=event.start,
-                    dtend=event.end,
-                    rrule=Recur.from_rrule(event.rrule),
-                    summary=f"{wa_name}Schedule {number}",
-                    uid=event.uid,
-                )
+        return [
+            CalendarEvent(
+                summary=program_event.program_id,
+                start=program_event.start.replace(
+                    tzinfo=ZoneInfo(str(dt_util.DEFAULT_TIME_ZONE))
```

Comment:
DEFAULT_TIME_ZONE is a `dt.tzinfo` already so i think this `ZoneInfo` wrapper is not needed or is a no-op. But also, it might be slight safer to take the tzinfo from the start_date object. Does this work? here and below/above

```suggestion
                    tzinfo=start_date.tzinfo
```

See https://developers.home-assistant.io/docs/core/entity/calendar#get-events for rationale

Reply:
As I tried it a had a mypy error, but not it works.